java/org/contikios/cooja/contikimote/interfaces/ContikiBeeper.java

```diff
@@ -86,9 +81,14 @@ public boolean isBeeping() {
   @Override
   public void doActionsAfterTick() {
     if (moteMem.getByteValueOf("simBeeped") == 1) {
-      this.setChanged();
-      this.notifyObservers(mote);
-
+      final var now = mote.getSimulation().getSimulationTime();
+      java.awt.EventQueue.invokeLater(() -> {
```

Comment:
Thanks for the update. One last thing: we should only invoke the dispatch thread if the HashMap is non-empty because it can only be non-empty when Cooja is visualized and not headless.

Reply:
Checking the HashMap can only be done in the AWT thread, but I added a check for visualized so AWT threads aren't started in headless mode.